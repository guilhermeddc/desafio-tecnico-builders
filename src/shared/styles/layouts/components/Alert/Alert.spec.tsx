import React from 'react';

import {render} from '@testing-library/react';

import {AlertComponent} from '.';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

jest.mock('shared/hooks', () => ({
  useCart: () => {
    return {
      cartQuantity: 0,
    };
  },
}));

describe('AlertComponent', () => {
  it('should render', () => {
    const {container} = render(<AlertComponent />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render with default props to type is equal success', () => {
    const setState = jest.fn();
    const useStateMock: any = () => [
      {
        type: 'success',
        isOpen: false,
        message: 'Success Message',
      },
      setState,
    ];

    jest.spyOn(React, 'useState').mockImplementation(useStateMock);

    const {container} = render(<AlertComponent />);
    expect(container).toBeInTheDocument();
  });

  it('should render with custom props to type is equal undefined', () => {
    const setState = jest.fn();
    const useStateMock: any = () => [
      {
        type: undefined,
        isOpen: false,
        message: 'Success Message',
      },
      setState,
    ];

    jest.spyOn(React, 'useState').mockImplementation(useStateMock);

    const {container} = render(<AlertComponent />);
    expect(container).toBeInTheDocument();
  });
});
