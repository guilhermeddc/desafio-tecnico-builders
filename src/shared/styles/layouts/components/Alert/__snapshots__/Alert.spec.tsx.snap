// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertComponent should render 1`] = `<div />`;
